src/ZenstruckFoundryBundle.php

```diff
@@ -341,6 +341,9 @@ private function configureInstantiator(array $config, ContainerBuilder $containe
             $container->getDefinition('.zenstruck_foundry.instantiator')
                 ->addMethodCall('alwaysForce', returnsClone: true)
             ;
+            $container->getDefinition('.zenstruck_foundry.maker.factory.generator')
+                ->addMethodCall('alwaysForce', returnsClone: true)
+            ;
```

Comment:
The error `You have requested a non-existent service ".zenstruck_foundry.maker.factory.generator".` happens because at this time, the service does not exist yet: see where is called `configureInstantiator()`, and where is called `$configurator->import('../config/makers.php');` few lines later.

Please, move this stuff at the end of the if: `if (isset($bundles['MakerBundle'])) {`  and you can set it like this:

```suggestion
            $container->getDefinition('.zenstruck_foundry.maker.factory.generator')
                ->setArgument('$forceProperties', $config['instantiator']['always_force_properties'])
            ;
```

I think this is enough if you add the default `false` is the constructor of `FactoryGenerator`

Reply:
I can't for the life of me get this to work....

I tried setting it a million different ways but I think once that's service is initialized and loaded you can't seem to change that value anymore

Reply:
big :facepalm: sorry, this "force `test` environment in constructor" stuff was a mistake

maybe we can force it with `APP_ENV=test` in  `.env`?

Reply:
but the problem still exists... something is strange here :thinking:

Reply:
ok, I found the culprit!
you should not call `[makeFactoryCommandTester](self::bootKernel(['environment' => 'always_force']);)` directly, but pass it through `makeFactoryCommandTester()`

Reply:
Oh yeah that does it, didn't even see that it booted the Kernel down there...